src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala

```diff
@@ -308,4 +307,18 @@ class BuiltinListFunctionsTest
     eval(" product(2,3,4) ") should be(ValNumber(24))
   }
 
+  "A joining function" should "return an empty string if the input list is empty" in {
+    eval(" joining([]) ") should be(ValString(""))
+  }
+
+  it should "return joined strings " in {
+    eval(""" joining(["foo","bar","baz"]) """) should be(ValString("foobarbaz"))
+    eval(""" joining(["foo","bar","baz"], "::") """) should be(
+      ValString("foo::bar::baz"))
+    eval(""" joining(["foo","bar","baz"], "::", "hello-")  """) should be(
+      ValString("hello-foo::bar::baz"))
+    eval(""" joining(["foo","bar","baz"], "::", "hello-", "-goodbye")  """) should be(
+      ValString("hello-foo::bar::baz-goodbye"))
+  }
```

Comment:
:wrench: We could split into different test cases for the different combinations of arguments.  

If we have different test cases then the combinations are more visible in the test spec. And it could be more transparent if somethings fails (i.e. following the rule "one assert per test case").

Reply:
ok!